app/queues/common/expiration-request.server.ts

```diff
@@ -0,0 +1,101 @@
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { prisma } from '~/db.server';
+
+import { addNotification } from '../notifications/notifications.server';
+import { deleteDnsRequest } from '../dns/delete-record-flow.server';
+import { RecordType } from '../notifications/expiration-notification.server';
+
+import type { ExpirationStatusPayload as ExpirationRequestPayload } from '../notifications/expiration-notification.server';
+
+const { EXPIRATION_REPEAT_FREQUENCY_MS, JOB_REMOVAL_FREQUENCY_MS } = process.env;
+const JOB_REMOVAL_INTERVAL = 7 * 24 * 60 * 60 * 1000;
+declare global {
+  var __expiration_request_init__: boolean;
+}
+// queue name
+const expirationRequestQueueName = 'expiration-request';
+
+// query to fetch expired records from database based on type
+const getExpiredRecords = (type: RecordType) => {
+  switch (type) {
+    case RecordType.DnsRecord:
+      return prisma.record.findMany({
+        where: {
+          expiresAt: {
+            lt: new Date(),
+          },
+        },
+        include: {
+          user: true,
+        },
+      });
+    case RecordType.Certificate:
+      logger.info(`TODO: return expired certificates from the db`);
+      break;
+    default:
+      throw new Error(`Invalid recordType: ${type}`);
+  }
+};
+
+// queue initialization
+const expirationRequestQueue = new Queue<ExpirationRequestPayload>(expirationRequestQueueName, {
+  connection: redis,
+});
+
+export function addExpirationRequest(type: RecordType) {
+  const jobName = `expirationRequestQueueName-${type}`;
+  return expirationRequestQueue.add(
+    jobName,
+    { type },
+    {
+      repeat: { every: Number(EXPIRATION_REPEAT_FREQUENCY_MS) || 24 * 60 * 60 * 1000 },
+      removeOnComplete: { age: Number(JOB_REMOVAL_FREQUENCY_MS) || JOB_REMOVAL_INTERVAL },
+      removeOnFail: { age: Number(JOB_REMOVAL_FREQUENCY_MS) || JOB_REMOVAL_INTERVAL },
+    }
+  );
+}
+
+// worker definition
+const expirationRequestWorker = new Worker<ExpirationRequestPayload>(
```

Comment:
What is the rationale in combining record and cert expiration? Wouldn't it be cleaner to make them separate?
Or do them both in one queue job (eliminating type)?

Either way would simplify the code quite a bit (especially the second)

Reply:
Agreed. It would be cleaner to do them in a single queue job.

Reply:
Let me know if [063424f](https://github.com/DevelopingSpace/starchart/pull/357/commits/063424f13bffda2bd529b0e1087f62a2f9d3b110) is the change you were requesting.

Reply:
What I suggest is to do away with `type`, i.e., this `case RecordType.DnsRecord:` would go away, and this worker would first process the DNS records, then would delete expired certificates from the DB, it would not take the "type" attribute at all

Reply:
Thanks for clarifying.